backend.py

```diff
@@ -0,0 +1,71 @@
+from flask import Flask, request, jsonify, send_file, render_template
+import subprocess
+import os
+import uuid
+
+app = Flask(__name__)
+
+@app.route('/')
+def index():
+    return render_template('frontend.html')  # Serve the frontend HTML
+
+@app.route('/upload', methods=['POST'])
+def upload_file():
+    if request.method == 'POST':
+        try:
+            # Retrieve front lens file and parameter
+            front_lens = request.files['frontLens']
+            bridge_length = request.form['bridgeLength']
+
+            # Generate unique IDs for the session or use the timestamp
+            unique_id = uuid.uuid4().hex
+            uploads_dir = 'uploads'
+            outputs_dir = 'outputs'
+            front_lens_filename = f"{unique_id}_front.dxf"
+            front_lens_path = os.path.join(uploads_dir, front_lens_filename).replace('\\', '/')
+            output_stl_path = os.path.join(outputs_dir, f"{unique_id}_frame.stl").replace('\\', '/')
+
+            # Ensure uploads and outputs directories exist
+            os.makedirs(uploads_dir, exist_ok=True)
+            os.makedirs(outputs_dir, exist_ok=True)
+
+            # Save the front lens file
+            front_lens.save(front_lens_path)
+
+            # Check if top lens file was provided, use default if not
+            if 'topLens' in request.files and request.files['topLens'].filename:
+                top_lens = request.files['topLens']
+                top_lens_filename = f"{unique_id}_top.dxf"
+                top_lens_path = os.path.join(uploads_dir, top_lens_filename).replace('\\', '/')
+                top_lens.save(top_lens_path)
+            else:
+                # Set to the default top lens file if not provided
+                top_lens_path = os.path.join(uploads_dir, 'RightLensTopView.dxf').replace('\\', '/')
+
+            # OpenSCAD executable path
+            openscad_path = r"C:\Program Files\OpenSCAD\openscad.exe"
```

Comment:
better to check OS and link to the correct file path

openscad_path = "/usr/local/bin/openscad"